yb-voyager/cmd/exportData.go

```diff
@@ -798,6 +807,200 @@ func getInitialTableList() (map[string]string, []sqlname.NameTuple) {
+			tuple = getNameTupleFromObjectName(obj)
+		default:
+			tuple, err = namereg.NameReg.LookupTableName(t)
+			if err != nil {
+				utils.ErrExit("lookup for table name failed: %s: %v", t, err)
+			}
+		}
+		fullTableList = append(fullTableList, tuple)
+	}
+	finalTableList = fullTableList
+
+	var partitionsToRootTableMap map[string]string
+	switch source.DBType {
+	case POSTGRESQL:
+		partitionsToRootTableMap = msr.SourceRenameTablesMap
+		if msr.SourceExportedTableListWithLeafPartitions == nil {
+			// In case of resumption this SourceExportedTableListWithLeafPartitions is not available (most likely in upgrade scenario)
+			//add the leafs in the finalTableList from the partitionsToRootTableMap
+			for leaf, root := range partitionsToRootTableMap {
+				if lo.ContainsBy(finalTableList, func(t sqlname.NameTuple) bool {
+					return t.AsQualifiedCatalogName() == root
+				}) {
+					defaultSchemaName, _ := getDefaultSourceSchemaName()
+					obj := sqlname.NewObjectNameWithQualifiedName(source.DBType, defaultSchemaName, leaf)
+					//get the name tupe via `getNAmeFromObj` function in case its leaf partition
+					leafTuple := getNameTupleFromObjectName(obj)
+					finalTableList = append(finalTableList, leafTuple)
+				}
+			}
+		}
+	case YUGABYTEDB:
+		//For the first run of export data from target we will fetch the leaf partitions from target and store them
+		if msr.TargetExportedTableListWithLeafPartitions == nil || msr.TargetRenameTablesMap == nil {
+			//Now add the leaf partitions to the stored table-list for the first run of export data from target
+			// as it will only have root table names and get the partitionsToRootTableMap
+			partitionsToRootTableMap, finalTableList, err = addLeafPartitionsInTableList(fullTableList, true)
+			if err != nil {
+				utils.ErrExit("failed to add the leaf partitions in table list: %w", err)
+			}
+		} else {
+			// On re-run we can re-use the stored one
+			finalTableList = lo.Map(msr.TargetExportedTableListWithLeafPartitions, func(t string, _ int) sqlname.NameTuple {
+				defaultSchemaName, _ := getDefaultSourceSchemaName()
+				obj := sqlname.NewObjectNameWithQualifiedName(source.DBType, defaultSchemaName, t)
+				//get the name tupe via `getNAmeFromObj` function in case its leaf partition
+				return getNameTupleFromObjectName(obj)
+			})
+			partitionsToRootTableMap = msr.TargetRenameTablesMap
+		}
+
+	}
+
+	//guardrails around the table-list in case of re-run
+
+	//TODO: fix later this registered list also contains sequence names right now
+	//so in case some include/exclude table names are not proper we will display this registeredList as valid table names
+	registeredListObjNames, err := namereg.NameReg.GetRegisteredTableList()
+	if err != nil {
+		utils.ErrExit("error getting registered list in name registry: %v", err)
+	}
+
+	registeredList := lo.Map(registeredListObjNames, func(obj *sqlname.ObjectName, _ int) sqlname.NameTuple {
```

Comment:
Why aren't we directly getting NameTuples from namereg.GetRegisteredTableList?

Reply:
As discussed, we need to handle nametuple look up in this code by checking if its a leaf of not